Src/AmrCore/AMReX_Bittree.cpp

```diff
@@ -0,0 +1,343 @@
+#include <AMReX_Bittree.H>
+#include <AMReX_ParallelDescriptor.H>
+
+#include <functional>
+
+namespace amrex {
+static constexpr int K1D = unsigned(AMREX_SPACEDIM>=1);
+static constexpr int K2D = unsigned(AMREX_SPACEDIM>=2);
+static constexpr int K3D = unsigned(AMREX_SPACEDIM>=3);
+
+
+bool btUnit::bcPeriodic[AMREX_SPACEDIM];
+
+/*
+NOTE: Bittree object is created in AmrMesh::MakeNewGrids (Real time)
+   with
+      `mesh = std::make_shared<BittreeAmr>(top,includes);`
+
+The functions here are called in the BT version of MakeNewGrids which has three steps:
+    1. Error Estimation and tagging - btTagging
+    2. Bitree's actual bitmap generated/updated - btRefine
+    3. AMReX updates grids based on bitree - btCalculateGrids
+*/
+
+
+/** New Bittree mesh is generated.
+  *
+  * This makes use of BT library functions and as well as routines adapted
+  * from Flash-X that enforce Octree nesting.
+  */
+int btUnit::btRefine( BittreeAmr* const mesh, std::vector<int>& btTags, MPI_Comm comm) {
+    // Tree before refinement. With only one rank, lnblocks = nblocks.
+    auto tree0 = mesh->getTree();
+    unsigned id0 = tree0->level_id0(0);
+    unsigned id1 = tree0->id_upper_bound();
+
+//--Mark leaves to be refined
+    for( unsigned id = id0; id < id1; ++id) {
+      if (btTags[id]==1) {
+        if(!tree0->block_is_parent(id)) {
+          mesh->refine_mark(id, true);
+        }
+      }
+    }
+    mesh->refine_reduce(comm);
+    mesh->refine_update();
+
+    btCheckRefine(mesh,btTags,comm);
+
+//--Mark derefinement (parents who will nodetype change to leaf)
+    for( unsigned id = id0; id < id1; ++id) {
+      if (btTags[id]==-1) {
+        if(tree0->block_is_parent(id)) {
+          mesh->refine_mark(id, true);
+        }
+      }
+    }
+    mesh->refine_reduce(comm);
+    mesh->refine_update();
+
+    btCheckDerefine(mesh,btTags,comm);
+
+    // return delta count
+    return static_cast<int>( mesh->delta_count() );
+}
+
+/** Creates new box arrays to match the new Bittree mesh.
+  */
+void btUnit::btCalculateGrids(BittreeAmr* const mesh, int lbase,
+                            Real time,int& new_finest,
+                            Vector<BoxArray>& new_grids,
+                            Vector<IntVect>& max_grid_size) {
+    auto tree1 = mesh->getTree(true);
+    int nlevs = tree1->levels();
+    new_finest = nlevs - 1;
+
+//--Calculate the new grid layout and distribution map based on Bittree
+    for(int lev=lbase; lev<=new_finest; ++lev) {
+        btCalculateLevel(mesh, lev, time, new_grids[lev],
+                         max_grid_size[lev]);
+    }
+}
+
+/** Creates a box array based on Bittree.
+  */
+void btUnit::btCalculateLevel(BittreeAmr* const mesh, int lev,
+                            Real time,
```

Comment:
```suggestion
                            Real /*time*/,
```
The CI is very strict on unused parameter.